src/wp-includes/block-template-utils.php

```diff
@@ -744,37 +749,23 @@ function _build_block_template_result_from_post( $post ) {
 	}
 	$parent_post = get_post( $post_id );
 
-	$terms = get_the_terms( $parent_post, 'wp_theme' );
-
-	if ( is_wp_error( $terms ) ) {
-		return $terms;
-	}
-
-	if ( ! $terms ) {
-		return new WP_Error( 'template_missing_theme', __( 'No theme is defined for this template.' ) );
-	}
-
-	$theme          = $terms[0]->name;
 	$template_file  = _get_block_template_file( $post->post_type, $post->post_name );
 	$has_theme_file = get_stylesheet() === $theme && null !== $template_file;
 
-	$origin           = get_post_meta( $parent_post->ID, 'origin', true );
-	$is_wp_suggestion = get_post_meta( $parent_post->ID, 'is_wp_suggestion', true );
-
 	$template                 = new WP_Block_Template();
 	$template->wp_id          = $post->ID;
 	$template->id             = $theme . '//' . $parent_post->post_name;
 	$template->theme          = $theme;
 	$template->content        = $post->post_content;
 	$template->slug           = $post->post_name;
 	$template->source         = 'custom';
-	$template->origin         = ! empty( $origin ) ? $origin : null;
+	$template->origin         = ! empty( $meta['origin'] ) ? $meta['origin'] : null;
 	$template->type           = $post->post_type;
 	$template->description    = $post->post_excerpt;
 	$template->title          = $post->post_title;
 	$template->status         = $post->post_status;
 	$template->has_theme_file = $has_theme_file;
-	$template->is_custom      = empty( $is_wp_suggestion );
+	$template->is_custom      = empty( $meta['is_wp_suggestion'] );
 	$template->author         = $post->post_author;
```

Comment:
I'm not sure `post_author` will always exist on `$post` especially as part of `$changes` in `inject_ignored_hooked_blocks_metadata_attributes`